Command line argument '--version' returns exit code 0 instead of 255

Bromeon opened the pull request:
Allows to detect whether printing the version was successful and makes integration with shell scripts straightforward.

Minor thing: `--version` is usually read by tools (CI, shell scripts, bindings) and at the moment, an extra newline is being printed, which requires trimming. On the other hand, it makes things a bit more readable for humans. Let me know if the current behavior is good as it is.

akien-mga commented on main/main.cpp:
```diff
@@ -673,6 +673,9 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 	packed_data->add_pack_source(zip_packed_data);
 #endif
 
+	// Default exit code, can be modified for certain errors
```
```suggestion
	// Default exit code, can be modified for certain errors.
```

akien-mga reviewed (commented):
Make sense, should also be added to `--help`. Other `goto error;` cases seem to all be proper invalid parameter errors so they're fine.

Bromeon committed:
Command line arguments '--version' and '--help' return exit code 0 instead of 255

Allows to detect whether those commands executed successfully, which makes integration with shell scripts/CI/bindings straightforward.

akien-mga reviewed (approved):
Looks good!

akien-mga commented:
Thanks!

Bromeon commented:
Thanks for the quick feedback! 🙂

akien-mga commented:
Cherry-picked for 3.5.

akien-mga commented:
This introduced a bug:
```
$ godot-git --version
4.0.alpha.custom_build.11a978b1b
ERROR: Condition "!_start_success" is true. Returning: false
   at: start (main/main.cpp:2096)
ERROR: Condition "!_start_success" is true.
   at: cleanup (main/main.cpp:2943)
$ echo $?
1
```
And doesn't actually achieve what was intended, since it still returns a failure code but further down.

I didn't pay close attention but this PR actually makes `setup()` return `OK`, which means that it will then proceed to `start()` instead of doing an early exit.

akien-mga commented:
Cherry-picked for 3.5.
